File: pypp_cli/test_dir/cpp/src/dicts/comprehensions.cpp
#include "dicts/comprehensions.h"
#include "py_dict.h"
#include "py_str.h"
#include "pypp_util/print.h"
#include <utility>

namespace me {
void dict_comprehension_fn() {
    pypp::print(pypp::PyStr("DICT COMPREHENSION RESULTS:"));
    pypp::PyDict<int, int> squares;
    for (int x = 0; x < 4; x += 1) {
        squares[std::move(x)] = x * x;
    }
    pypp::print(pypp::PyStr(std::format("Squares: {}", squares)));
}

} // namespace me

File: test_dir/src/dicts/comprehensions.py
from pypp_python import mov


def dict_comprehension_fn():
    print("DICT COMPREHENSION RESULTS:")
    # Dict comprehension to create a dictionary of squares
    squares: dict[int, int] = {mov(x): x * x for x in range(4)}
    print(f"Squares: {squares}")
